test/CMakeLists.txt

```diff
@@ -57,11 +60,7 @@ endif()
 
 enable_testing()
 
-# Note: doctest and similar testing frameworks can automatically configure CMake tests For other
-# testing frameworks add the tests target instead: ADD_TEST(GreeterTests GreeterTests)
-
-include(${doctest_SOURCE_DIR}/scripts/cmake/doctest.cmake)
-doctest_discover_tests(GreeterTests)
+add_test(NAME greeterTests COMMAND GreeterTests)
```

Comment:
As discussed before, I prefer using `doctest_discover_tests` as it propagates doctest test cases to ctest. Under normal circumstances the `doctest_SOURCE_DIR` should always be defined as it's the standard behaviour of CPM.cmake to always download from source. Unless I'm mistaken it seems that you have `CPM_USE_LOCAL_PACKAGES` enabled on your system, an advanced feature that should be used with caution. Specifically, it's not recommended for tests, as it makes builds system-dependent (as we have noticed through the error).

I'd recommend either unsetting the flag `CPM_USE_LOCAL_PACKAGES` for development on your system, or modifying the starter for your personal projects only, if you prefer the flag to be enabled. For normal use-cases I believe we can safely assume that `CPM_USE_LOCAL_PACKAGES` is not enabled and the command will therefore work on all systems.

Reply:
As I noted before, it does not work if dockets is installed on host and this installed package is use.
It is an option of CPM.cmake I like.

I already noted, with a real use case, we're many small project are build together, it does't scale!
I do not want to build big libraries like boost, folly, ACE, ... every time we use them.

Reply:
That's fine and you're welcome to adjust it for your starter / projects. I personally prefer to have reproducible builds (as promised by the template) and use projects such as [ccache](https://ccache.dev) to avoid long compilation times for large dependencies. This has the advantage of being able to using such library functions and avoiding special cases without worrying which version `find_package` may have or have not picked up.

Reply:
what is the benefit to use the `doctest_discover_tests(GreeterTests)`

and how can we give arguments like `-s` to you `doctest` main?

Reply:
- what is the benefit to use the doctest_discover_tests(GreeterTests)

It allows ctest to generate much better output as it is aware of all test cases.

- and how can we give arguments like -s to you doctest main?

You can still manually launch `./build/test/GreeterTests` with arbitrary arguments as before.